### PR TITLE
Add super user database connections

### LINKED ISSUE #38: Need mechanism to request db superuser connection
There are some schema changes that can only be done by a db superuser (_e.g._ creating functions in an untrusted language, like plpythonu) However, it's bad practice to make all your connections as superuser. I propose a superuser() context manager script, over in https://github.com/Connexions/cnx-db/pull/23 and @pumazi suggested it should probably properly live in db-migrator proper. If so, I'd add some sort of option to provide a different super-user name (defaulting to postgres) at runtime.

### PR DESCRIPTION
**Pull request by karenc**
Add super_user() context manager that can be used in migrations when a
database super user connection is necessary.

Close #38

### CODE REVIEW

**Comment by karenc** [comment]
@reedstrm

**Comment by reedstrm** [comment]
If the above works (do super first) then this is good to go, with a note in the template, maybe?

**Comment by karenc** [comment]
The only problem I see with this approach is it is a new connection, if what runs in the superuser() block depends on what happens before it, it's not going to be committed to the database yet (I think?). We might have a partially migrated database as well, because the superuser() stuff might succeed while the code after that might fail and not get committed.

But that's something to keep in mind when writing the migrations, implementing superuser() in db-migrator isn't a problem.

(From https://github.com/karenc/db-migrator/issues/38#issuecomment-322406117)

File: dbmigrator/tests/test_utils.py [hunk 1]
```diff
@@ -36,6 +36,38 @@ def tearDown(self):
...
+                    super_cursor.execute('CREATE TABLE super_table (id TEXT)')
+                    super_cursor.execute(
+                        'GRANT ALL PRIVILEGES ON TABLE super_table TO travis')
+                    super_cursor.execute(
+                        "INSERT INTO super_table VALUES ('62')")
+
+                cursor.execute('SELECT current_user')
+                self.assertEqual('travis', cursor.fetchone()[0])
+
+                cursor.execute('SELECT * FROM super_table')
```
Comment (karenc) [inline dbmigrator/tests/test_utils.py]: @reedstrm I added this to make sure the table created in the context manager can be accessed outside of it.  It appears to be working.  Is there a specific case when it doesn't work?

Comment (reedstrm) [reply dbmigrator/tests/test_utils.py]: Hmm, I'll look into this. Was pretty sure I had run into it.

**Comment by karenc** [comment]
[CNX database privileges issues](https://trello.com/c/1oTABKzL/278-cnx-database-privileges-issues)

**Pull request merged**